sematic/api/endpoints/request_parameters.py

```diff
@@ -294,6 +296,37 @@ def _none_if_empty(name: str) -> Optional[str]:
 
 
 def jsonify_error(error: str, status: HTTPStatus):
+    is_warning = False
+
+    # Since jsonify_error is explicitly called to handle the
+    # error, there is no uncaught exception to be handled by
+    # the "catch all" exception handler & logger. This ensures
+    # that we generate a log message even for these explicitly
+    # handled errors, so we can triage more easily.
+    if status is None:
+        logger.error("Attempting to create an error without a code: %s", error)
+    elif 400 <= status.value < 500:
+        logger.warning("Bad request error: '%s'", error, stack_info=True)
+        is_warning = True
+    else:
+        logger.exception("Server error: '%s'", error)
+
+    # The above prints the error summary and stack to *this* call.
+    # We also want to show the stack trace for the root cause exception,
+    # if there indeed was one.
+    exception_type, exception, trace = sys.exc_info()
+    if exception is None:
+        logger.warning("No exception cause")
+    else:
+        for line in traceback.format_exception(
+            etype=exception_type, value=exception, tb=trace
+        ):
+            line = line[:-1]  # strip terminal newline
+            if is_warning:
+                logger.warning(line)
+            else:
+                logger.error(line)
```

Comment:
This line is unreachable because all call sites where we have an error in the stack trace are 500s.